### PR TITLE
prevent-delete-private-dns-zone

### PR DESCRIPTION
**Pull request by sdolgin**
This policy definition prevents the deletion of private DNS zones in Azure, using the DenyAction effect. It serves as an alternative to using CanNotDelete resource locks to prevent the deletion of Private DNS Zones that are used with private endpoints. It's important to note that this policy does not prevent the deletion of child resources (i.e. the DNS Zone records themselves).

### CODE REVIEW

**Comment by mrajess** [comment]
Hello! Thank you for your contribution. We recently updated our [contribution guide](https://github.com/Azure/Community-Policy/blob/master/CONTRIBUTING.md) to include a few new guidelines, of which you're presently missing, but they should be quick fixes. [Here's an example](https://github.com/Azure/Community-Policy/blob/master/Policies/Authorization/deploy-or-audit-for-role-assignment-at-sub-scope/azurepolicy.json) of a Policy I just submitted using the new guidelines.

In short, could you include:
- parameterize the effect
- wrap your current policy file in another set of curly braces {}
     - in the new curly braces include the policy name and have it generated as a guid
     - include the metadata field and include the version number and a category
     - include the resource type
     - include the api version

If you could just do that real quick I'll go ahead and approve your PR. Thanks again for your contribution!

**Review by techlake — CHANGES_REQUESTED** [review]
You're missing a name (GUID). Please follow [Contribution Guidance](https://github.com/Azure/Community-Policy/blob/master/CONTRIBUTING.md). You can use our handy script to fix your submission: https://github.com/Azure/Community-Policy/blob/master/Submit-PolicyDefinitionFile.ps1

**Comment by sdolgin** [comment]
Hi there @mrajess 
  I think I got the guidelines addressed now.  Please let me know if there are any remaining issues with this contribution.
